src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -177,9 +201,27 @@ export class SharedCredentialsProvider implements CredentialsProvider {
         }
     }
 
+    private validateSourcedCredentials(): string | undefined {
+        if (hasProfileProperty(this.profile, SHARED_CREDENTIAL_PROPERTIES.SOURCE_PROFILE)) {
```

Comment:
Does this explicitly error either through the JS SDK or the CLI? If not, I'd just default to that behavior.

Reply:
The docs for credentials explicitly state that both cannot be set.